src/main/java/nextstep/domain/Ladder.java

```diff
@@ -19,11 +29,24 @@ private List<Integer> play(Lines lines, int userCount) {
                 .collect(Collectors.toList());
     }
 
-    public int getResult(int position) {
-        return results.get(position);
+    private Map<Username, String> match(List<Integer> ladderResults,
+                                               List<String> results,
+                                               Usernames usernames) {
+        Map<Username, String> finalResults = new LinkedHashMap<>();
+        List<String> inputResults = ladderResults.stream()
+                .map(results::get)
+                .collect(Collectors.toList());
+
+        for (int index = 0; index < inputResults.size(); index++) {
+            finalResults.put(usernames.getUsernames().get(index), inputResults.get(index));
```

Comment:
```suggestion
            finalResults.put(usernames.get(index), inputResults.get(index));
```
이런 것들도 점으로 이어서 하는 것 보다는 일급컬렉션 내에서 바로 처리할 수 있을 것 같네요